recipes/cpr/all/conanfile.py

```diff
@@ -120,42 +140,98 @@ def _configure_cmake(self):
+            self.output.warn("with_openssl is deprecated. Please use the with_ssl option.")
+        elif self.options.get_safe("with_winssl") != "deprecated":
+            self.output.warn("with_winssl is deprecated. Please use the with_ssl option.")
+
+    # Get the configured ssl library
+    def _get_ssl_library(self):
+        ssl_library = str(self.options.get_safe("with_ssl"))
+        # These must check for True so that we don't take "deprecated" to be truthy
+        if self.options.get_safe("with_openssl") == True:
+            return "openssl"
+        elif self.options.get_safe("with_winssl") == True:
+            return "winssl"
+        elif not self._can_auto_ssl and ssl_library == CprConan._AUTO_SSL:
+            if self._supports_openssl:
+                self.output.info("Auto SSL is not available below version 1.6.0. Falling back to openssl")
+                return "openssl"
+            elif not self._supports_openssl:
+                self.output.info("Auto SSL is not available below version 1.6.0, and openssl not supported. Disabling SSL")
+                return CprConan._NO_SSL
+        elif ssl_library == CprConan._AUTO_SSL and tools.Version(self.version) in ["1.6.0", "1.6.1"] and tools.is_apple_os(self.settings.os):
+           #  https://github.com/whoshuu/cpr/issues/546
+            self.output.info("Auto SSL is not available on macOS with version 1.6.0/1.6.1, and openssl not supported. Disabling SSL")
+            return CprConan._NO_SSL
+
+        return ssl_library
+
     def validate(self):
-        if not self._uses_valid_abi_and_compiler:
-            raise ConanInvalidConfiguration("Cannot compile CPR/1.6.0 with libstdc++ on clang < 9")
+        SSL_FAILURE_MESSAGES = {
+            "openssl": "OpenSSL is not supported on macOS or on CPR versions < 1.5.0",
+            "darwinssl": "DarwinSSL is only supported on macOS and on CPR versions >= 1.6.1",
+            "winssl": "WinSSL is only on Windows and on CPR versions >= 1.5.1",
+            CprConan._AUTO_SSL: "Automatic SSL selection is only available on CPR versions >= 1.6.0"
+        }
 
-        if self.options.get_safe("with_openssl", False) and tools.is_apple_os(self.settings.os):
-            # https://github.com/whoshuu/cpr/issues/546
-            raise ConanInvalidConfiguration("cpr cannot be built on macOS with openssl")
+        self._print_deprecation_messages()
+        if not self._uses_valid_abi_and_compiler:
+            raise ConanInvalidConfiguration("Cannot compile cpr/1.6.0 with libstdc++ on clang < 9")
 
-        if self.options.get_safe("with_winssl", False) and self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("cpr only supports winssl on Windows")
+        ssl_library = self._get_ssl_library()
+        if not self._supports_ssl_library(ssl_library):
+            raise ConanInvalidConfiguration(
+                f"Invalid SSL selection for the given configuration: {SSL_FAILURE_MESSAGES[ssl_library]}"
```

Comment:
We are trying typing to maintain support for older python versions, could you please use `"".format()` instead please?

Reply:
Ah shoot - wasn't thinking about that. I've been in the `f-string` mindset for too long :sweat_smile: